### PR TITLE
Update Router

### PR DESCRIPTION
**Pull request by oTkPoBeHuE**

### CODE REVIEW

File: src/components/Router/Route.tsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+
+export type RouteProps = {
+    path: string;
+    regex?: boolean;
```
Comment (ledamint) [inline src/components/Router/Route.tsx]: Maybe path: string | RegExp; ?

File: src/components/Router/Router.tsx [hunk 2]
```diff
@@ -20,17 +17,13 @@ export function Router(props: RouterProps) {
     });
 
     const childrenArray = React.Children.toArray(props.children) as React.ReactElement<RouteProps>[];
-    const child = childrenArray.find((child) => {
+    const child = childrenArray.filter((child) => {
```
Comment (ledamint) [inline src/components/Router/Router.tsx]: I guess we should use find

File: src/components/Router/Route.tsx (hunk 1 above)
Comment (oTkPoBeHuE) [reply src/components/Router/Route.tsx]: fixed

File: src/components/Router/utils.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import { Route, RouteProps } from './Route';
+
+type PatchOptions = { pattern?: string | RegExp };
+
+export const matchRoute = (pathname: string, options: PatchOptions) => {
```
Comment (ledamint) [inline src/components/Router/utils.ts]: `export const matchPattern = (path: string, pattern: string | RegExp) => {`

File: src/components/Router/utils.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import { Route, RouteProps } from './Route';
+
+type PatchOptions = { pattern?: string | RegExp };
```
Comment (ledamint) [inline src/components/Router/utils.ts]: is excess I guess

File: src/components/Router/utils.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    }
+
+    if (pattern instanceof RegExp) {
+        return pattern.test(pathname);
+    }
+
+    return pattern === pathname;
+};
+
+export const matchChild = (url: string) => (child: React.ReactElement<RouteProps>) => {
```
Comment (ledamint) [inline src/components/Router/utils.ts]: `export const matchRoute = (url: path)`

**Pull request merged**